frontend/apps/reader/modules/readerlink.lua

```diff
@@ -895,72 +835,63 @@ function ReaderLink:onGoToPageLink(ges, use_page_first_link, allow_footnote_popu
-                if use_page_first_link then
-                    -- links may not be in the order they are in the page, so let's
-                    -- find the one with the smallest start_y.
-                    if first_start_y == nil or link["start_y"] < first_start_y then
-                        selected_link = link
-                        first_start_y = link["start_y"]
-                    end
-                else
-                    if link["segments"] then
-                        -- With segments, each is a horizontal segment, with start_x < end_x,
-                        -- and we should compute the distance from gesture position to
-                        -- each segment.
-                        local segments_max_y = -1
-                        local link_is_shortest = false
-                        local segments = link["segments"]
-                        for i=1, #segments do
-                            local segment = segments[i]
-                            local segment_dist
-                            -- Distance here is kept squared (d^2 = diff_x^2 + diff_y^2),
-                            -- and we compute each part individually
-                            -- First, vertical distance (squared)
-                            if pos_y < segment.y0 then -- above the segment height
-                                segment_dist = math.pow(segment.y0 - pos_y, 2)
-                            elseif pos_y > segment.y1 then -- below the segment height
-                                segment_dist = math.pow(pos_y - segment.y1, 2)
-                            else -- gesture pos is on the segment height, no vertical distance
-                                segment_dist = 0
-                            end
-                            -- Next, horizontal distance (squared)
-                            if pos_x < segment.x0 then -- on the left of segment: calc dist to x0
-                                segment_dist = segment_dist + math.pow(segment.x0 - pos_x, 2)
-                            elseif pos_x > segment.x1 then -- on the right of segment : calc dist to x1
-                                segment_dist = segment_dist + math.pow(pos_x - segment.x1, 2)
-                            -- else -- gesture pos is in the segment width, no horizontal distance
-                            end
-                            if shortest_dist == nil or segment_dist < shortest_dist then
-                                selected_link = link
-                                shortest_dist = segment_dist
-                                link_is_shortest = true
-                            end
-                            if segment.y1 > segments_max_y then
-                                segments_max_y = segment.y1
-                            end
+            -- link.uri may be an empty string with some invalid links: ignore them
+            if link["section"] or (link["uri"] and link["uri"] ~= "") then
```

Comment:
Seems easier to just write:
```suggestion
            if link.section or (link.uri and link.uri ~= "") then
```